test/unit/sourceset.test.js

```diff
@@ -0,0 +1,479 @@
+/* eslint-env qunit */
+import videojs from '../../src/js/video.js';
+import document from 'global/document';
+import window from 'global/window';
+import log from '../../src/js/utils/log.js';
+import sinon from 'sinon';
+
+const Html5 = videojs.getTech('Html5');
+const wait = 1;
+let qunitFn = 'module';
+
+if (!Html5.canOverrideAttributes()) {
+  qunitFn = 'skip';
+}
+
+const oldMovingMedia = Html5.prototype.movingMediaElementInDOM;
+const validateSource = function(assert, player, sources, checkMediaElSource = true) {
+  const tech = player.tech_;
+  const mediaEl = tech.el();
+
+  if (checkMediaElSource) {
+    assert.equal(mediaEl.src, sources[0].src, 'mediaEl.src is correct');
+    assert.equal(mediaEl.getAttribute('src'), sources[0].src, 'mediaEl attribute is correct');
+    assert.equal(tech.src(), sources[0].src, 'tech is correct');
+  }
+};
+
+const setupEnv = function(env, testName) {
+  env.fixture = document.getElementById('qunit-fixture');
+
+  if (testName === 'change video el' || testName === 'change audio el') {
+    Html5.prototype.movingMediaElementInDOM = false;
+  }
+
+  env.sourcesets = 0;
+  env.hook = (player) => player.on('sourceset', () => env.sourcesets++);
+  videojs.hook('setup', env.hook);
+
+  if ((/audio/i).test(testName)) {
+    env.mediaEl = document.createElement('audio');
+  } else {
+    env.mediaEl = document.createElement('video');
+  }
+  env.testSrc = {
+    src: 'http://vjs.zencdn.net/v/oceans.mp4',
+    type: 'video/mp4'
+  };
+  env.sourceOne = {src: 'http://example.com/one.mp4', type: 'video/mp4'};
+  env.sourceTwo = {src: 'http://example.com/two.mp4', type: 'video/mp4'};
+
+  env.mediaEl.className = 'video-js';
+  env.fixture.appendChild(env.mediaEl);
+};
+
+const setupAfterEach = function(totalSourcesets) {
```

Comment:
this is all the code from the `afterEach`, everything was the same except the value set for `totalSourcesets`, so, just made a HOF.